### PR TITLE
Build Shopping Results page by scraping Google Shopping

### PR DESCRIPTION
**Pull request by ralucatudor**
Create a separate page for the shopping results - containing the products returned from scraping Google Shopping results page for a hard-coded search query.

The work has been split between Saakshi and Raluca:
-> Saakshi handled the back-end part. Made a hard-coded query and fetched the results from Google Shopping Search (without any adverts) using Jsoup. Returned the SERP (Search Engine Results Page) - in the form of a HTML page (of the Google Shopping search query). 
-> Given the HTML, Raluca extracted the information needed to integrate the products into the webpage. Created a separate page for the results, which includes an empty wrapper for the products - inside the wrapper, the product containers are inserted.

### CODE REVIEW

File: src/main/webapp/shopping-results.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    // Get info about the current product by extracting from the HTML element.
+
+    const productImageLink = $('.oR27Gd > img', currentProductHTML).attr('src');
+
+    let productLink = $('.rgHvZc > a', currentProductHTML).attr('href');
+
+    // Fix product link - if the URL starts with '/url?q=', the URL redirection will not work.
+    const wrongStartOfLink = '/url?q=';
+    // Therefore delete the start if this is the case, for the redirection to successfully work.
+    if (productLink.substring(0, wrongStartOfLink.length) == wrongStartOfLink) {
```
Comment (saakshisaxena) [inline src/main/webapp/shopping-results.js:47]: I think I'll be better if we use a `===` here. What say?
```suggestion
    if (productLink.substring(0, wrongStartOfLink.length) === wrongStartOfLink) {
```

File: src/main/webapp/shopping-results.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      productLink = productLink.substring(wrongStartOfLink.length);
+    }
+
+    // Get the title as HTML instead of text, in order to keep the <b> tags.
+    const productTitle = $('.rgHvZc > a', currentProductHTML).html();
+
+    const productPrice = $('.dD8iuc > .HRLxBb', currentProductHTML).text()
```
Comment (saakshisaxena) [inline src/main/webapp/shopping-results.js:54]: Nit: semicolon missing
```suggestion
    const productPrice = $('.dD8iuc > .HRLxBb', currentProductHTML).text();
```

**Review by saakshisaxena — COMMENTED** [review]
Kudos to finishing a main step in our project  🎉 

Just a couple of nits for review.

File: src/main/webapp/shopping-results.js (hunk 1 above)
Comment (ralucatudor) [reply src/main/webapp/shopping-results.js:47]: Indeed, thanks!

**Pull request closed without merging**